### PR TITLE
[s] Adds camera bug camera network to restricted_camera_networks

### PR DESCRIPTION
**Pull request by Adrer**
## What Does This PR Do
Adds the camera bug camera network to restricted_camera_networks when it is initialized to prevent the AI from accessing the network.

## Why It's Good For The Game
Not particularly stealthy if the AI can jump onto the network as well.

## Testing
Spawn in a camera bug kit.
Turn into the AI.
Camera bug network does not appear in the jump to network list.

## Changelog
:cl:
fix: AI can no longer access camera bug networks.
/:cl:

### CODE REVIEW

**Review by GDNgit — APPROVED** [review]
I'd suggest caching this but that's just extra

**Comment by farie82** [comment]
Can you also remove it on destroy?

File: code/game/objects/items/devices/camera_bug.dm [hunk 1]
```diff
@@ -25,6 +25,7 @@
 	integrated_console = new(src)
 	integrated_console.parent = src
 	integrated_console.network = list("SS13", "camera_bug[UID()]")
+	GLOB.restricted_camera_networks += "camera_bug[UID()]"
```
Comment (Contrabang) [inline code/game/objects/items/devices/camera_bug.dm:28]: I don't think this actually prevents the AI from seeing them, it just prevents them from jumping to it from what I can tell.

**Review by Contrabang — APPROVED** [review]
Code LGTM

File: code/game/objects/items/devices/camera_bug.dm [hunk 2]
```diff
@@ -25,6 +25,7 @@
...
 
 /obj/item/camera_bug/Destroy()
 	QDEL_NULL(integrated_console)
```
Comment (lewcc) [inline code/game/objects/items/devices/camera_bug.dm:31]: Headcoders are in favor of making this change, should be a quick one.
```suggestion
	GLOB.restricted_camera_networks -= "camera_bug[UID()]"
	QDEL_NULL(integrated_console)
```

**Review by lewcc — DISMISSED** [review]

File: code/game/objects/items/devices/camera_bug.dm (hunk 2 above)
Comment (Contrabang) [reply code/game/objects/items/devices/camera_bug.dm:31]: 99% sure this means that if this happens, and the camera bug item is deleted, it means that the AI will be able to view the  hidden camera network, Wouldn't it? If headcoders prefer that, thats fine, I just want to make sure all the side effects are properly communicated.

Comment (lewcc) [reply code/game/objects/items/devices/camera_bug.dm:31]: I see what you mean, I'll bring that up

Comment (lewcc) [reply code/game/objects/items/devices/camera_bug.dm:31]: Yeah, it seems like we don't need to worry about this as per discussions

**Review by S34NW — APPROVED** [review]

**Pull request merged**